eis_toolkit/statistical_analyses/statistical_testing.py

```diff
@@ -0,0 +1,85 @@
+import pandas as pd
+from beartype import beartype
+from beartype.typing import Literal, Optional
+from scipy.stats import anderson, chi2_contingency, shapiro
+
+from eis_toolkit import exceptions
+
+
+def _statistical_tests(
+    data: pd.DataFrame, target_column: str, method: Literal, min_periods: Optional[int], delta_degrees_of_freedom: int
+) -> dict:
+
+    statistics = {}
+    normality = {}
+
+    for column in data.columns:
+        if data[column].dtype != float:  # Categorical variables
+            if column != target_column:
+                contingency_table = pd.crosstab(data[target_column], data[column])
+                chi_square, p_value, degree_of_freedom, _ = chi2_contingency(contingency_table)
+                statistics[column] = {
+                    "chi-square": chi_square,
+                    "p-value": p_value,
+                    "degrees of freedom": degree_of_freedom,
+                }
+        else:  # Numerical variables
+            normality[column] = {"shapiro": shapiro(data[column]), "anderson": anderson(data[column], "norm")}
+
+    if normality:
+        statistics["normality"] = normality
+
+    # Create subset of numerical variables only
+    numerical_columns = data.select_dtypes(include=["float"])
+    if not numerical_columns.empty:
+        statistics["correlation matrix"] = numerical_columns.corr(method=method, min_periods=min_periods)
+        statistics["covariance matrix"] = numerical_columns.cov(min_periods=min_periods, ddof=delta_degrees_of_freedom)
+
+    return statistics
+
+
+@beartype
+def statistical_tests(
+    data: pd.DataFrame,
+    target_column: str,
```

Comment:
I thought about the inputs for a while, and I think it could be better if the inputs were simply two arrays of same size (Numpy array or other sequence). If there are multiple pairings the user wishes to test, they could call this function multiple times.

Reply:
Sorry, I don't quite follow. What would the other array be for?